pdpserver/server/ctrl_storage.go

```diff
@@ -0,0 +1,60 @@
+package server
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/infobloxopen/themis/pdp"
+)
+
+const missingStorageMsg = `"Server missing policy storage"`
+
+type storageHandler struct {
+	s *Server
+}
+
+func (handler *storageHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	var (
+		depth int64
+		err   error
+	)
+
+	// parse depth
+	queryOpt := r.URL.Query()
+	if depthOpt, ok := queryOpt["depth"]; ok {
+		depthStr := depthOpt[0]
+		depth, err = strconv.ParseInt(depthStr, 10, 64)
```

Comment:
We don't want to get negative depth. Please use `strconv.ParseUint`. However uint64 → int can overflow int. I'd put check `if depth > math.MaxInt32 { return 400 }`.

Reply:
I think there's already a negative depth check in place:
`if depth < 0 { return newMarshalInvalidDepthError(depth) }`

However, you raise a good point. Why not make depth in MarshalWithDepth typed as uint?
We can also use strconv.ParseUint(depthStr, 10, 32) to enforce 32 bit length when parsing.
Do you forsee any problems with this strategy?

Reply:
Yes, we check the depth at library level but it leads to 500 error while semantically it's 400. You need to export MarshalInvalidDepthError and check for it or put the validation I wrote about.

`strconv.ParseUint(depthStr, 10, 32)` looks good to me.